vscode/src/svelte/components/file/FileTreeElement.ts

```diff
@@ -0,0 +1,75 @@
+import { writable } from 'svelte/store';
+import { WorkspaceGraphElement, FileInclusionStatus, MentatArgs } from '../../../types/globals';
+
+export class FileTreeElement {
+    public children: FileTreeElement[] | undefined;
+    // A svelte-subscribable store that stores broadcasts changes in status
+    public statusStore = writable(FileInclusionStatus.notIncluded);
+    public status: FileInclusionStatus | undefined; // A locally-accessible cache of status
+
+    constructor(
+        public file: WorkspaceGraphElement, // File metadata
+    ) {
+        if (this.file.children) {
```

Comment:
For me there's an ~11s time to load after clicking the refresh button on this project (with python and node dependencies installed). Moving this to handleClick and loading files lazily cut it to ~3s.

Reply:
Aha - great idea. I've now done something a bit different but same principle.

Reply:
Your approach is really clean. Nice!